src/pad.jl

```diff
@@ -0,0 +1,104 @@
+"""
+    PaddedDiskArray <: AbstractDiskArray
+
+    PaddedDiskArray(A, padding; fill=zero(eltype(A)))
+
+An `AbstractDiskArray` that adds padding to the edges of the parent array.
+This can help changing chunk offsets or padding a larger than memory array
+before windowing operations.
+
+# Arguments
+
+- `A`: The parent disk array.
+- `padding`: A tuple of `Int` lower and upper padding tuples, one for each dimension.
+
+# Keywords
+
+- `fill=zero(eltype(A))`: The value to pad the array with.
+"""
+struct PaddedDiskArray{T,N,A<:AbstractArray{T,N},C,F<:T} <: AbstractDiskArray{T,N}
+    parent::A
+    padding::NTuple{N,Tuple{Int,Int}}
+    fill::F
+    chunks::C
+end
+function PaddedDiskArray(A::AbstractArray{T,N}, padding::NTuple{N,Tuple{Int,Int}};
+    fill=zero(eltype(A)),
+) where {T,N}
+    chunks = GridChunks(map(_pad_offset, eachchunk(A).chunks, padding, size(A)))
+    PaddedDiskArray(A, padding, fill, chunks)
+end
+
+function _pad_offset(c::RegularChunks, (low, high), s)
+    chunksize = c.cs
+    offset = low > 0 ? c.offset - low + chunksize : c.offset
+    size = c.s + low + high
+    return RegularChunks(chunksize, offset, size)
+end
+function _pad_offset(c::IrregularChunks, (low, high), s)
```

Comment:
It is not really obvious to me what the code should do here. Currently this adds an extra chunk at the beginning for the padded values. Should the values padded at the end be in a new chunk as well? Or should padded values simply be lumped into the first and last chunk. However, something is still wrong with the implementation, see this example:

````julia
c2 = IrregularChunks(chunksizes = [10, 10, 20, 30, 40])
_pad_offset(c2, (5,5), 100)
````

which returns chunks only up to 105 while the resulting array should have length 120.

Reply:
Thanks, IrregularChunks needs more work

Reply:
fixed